src/rideTask.cpp

```diff
@@ -0,0 +1,123 @@
+#include "rideTask.hpp"
+#include "Particle.h"
+#include <time.h>
+#include "cli/conio.hpp"
+#include "consts.hpp"
+#include "product.hpp"
+#include "system.hpp"
+#include "sleepTask.hpp"
+#include "util.hpp"
+#include "vers.hpp"
+#include "scheduler.hpp"
+#include "ensembles.hpp"
+
+static void RIDE_setFileName(system_tick_t startTime){return;}
+
+DeploymentSchedule_t deploymentSchedule[] = 
+{ 
+    {SS_ensembleAFunc, SS_ensembleAInit, 1, 100, 2000, UINT32_MAX, 0, 0, 0, 
+                                                        nullptr,400,(char) 65},
+    {SS_ensembleBFunc, SS_ensembleBInit, 1, 600, 2000, UINT32_MAX, 0, 0, 0, 
+                                                        nullptr,200,(char) 66}, 
+    {SS_ensembleCFunc, SS_ensembleCInit, 1, 900, 2000, UINT32_MAX, 0, 0, 0, 
+                                                        nullptr,600,(char) 67}, 
+    {nullptr, nullptr, 0, 0, 0, 0, 0, 0, 0, nullptr}
+};
+uint32_t rollingMean(DeploymentSchedule_t* event, int duration)
+{
+
+    int next = (duration - (int)event->maxDuration) /
+    ((int)event->measurementCount + 1);
+    return (uint32_t) (event->maxDuration + next);
+}
+
+STATES_e RideTask::run(void)
+{
+    
+    SF_OSAL_printf("\nStarting RideTask::run\n");
+    DeploymentSchedule_t* pNextEvent = NULL;
+    system_tick_t nextEventTime;
+    int counter = 0;
+    
+    uint32_t d;
+    SF_OSAL_printf("\nDeployment started at %u\n",millis());
+    while(1)
+    {
+        
+        RIDE_setFileName(this->startTime);
+        
+        SCH_getNextEvent(deploymentSchedule, &pNextEvent, &nextEventTime);
+        
+        d = 0;
+        int m = counter % 17;
+        if(m == 3)
+        {
+            d = 200; 
+        }
+        else if (m == 8 || m == 13)
+        {
+            d = 800;
+        }
+        while(millis() < nextEventTime)
+        {
+            continue;
+        }
```

Comment:
Would it make more sense to compute the time to the next event, then simply execute `delay`?